services/core-api/app/api/securities/resources/bond.py

```diff
@@ -89,6 +89,7 @@ def get(self, bond_guid):
     def put(self, bond_guid):
         #remove the amount from the request if it exists as it should not be editable.
         temp_bond = Bond.find_by_bond_guid(bond_guid)
+        temp_bond.save_bond_history()
```

Comment:
What if the below operations fail? We'll have incorrectly saved bond history for a failed edit operation. I think this should happen at the end after the bond has been successfully modified.

Reply:
If I want to save the history the way that I am with a class method the bond history save has to happen before the bond is modified

Reply:
I guess an option is to then delete this `temp_bond` object in the `catch` block.

Reply:
Temp_bond is just the bond before we modify it. So I dont want to delete it, the function lives on the bond model so it takes self and marshals it then loads it into the history table.

Reply:
I understand. Do we want to create a bond history record if the PUT operation fails?

Reply:
I have changed this and it should be good now if you can take another look